src/sync.rs

```diff
@@ -199,12 +209,29 @@ impl LatestState {
 pub fn sync(
     stdout: &mut StandardStream,
     info_color_spec: ColorSpec,
-    mail_dir: PathBuf,
     args: Args,
     config: Config,
 ) -> Result<(), Error> {
+
+    let state_dir = match &config.state_dir {
+        Some(ref dir) => dir.clone(),
+        _ => {
+            let project_dirs = ProjectDirs::from("sh.eliza", "", "mujmap").unwrap();
```

Comment:
Can we avoid instantiating a `ProjectDirs` twice?

Reply:
Depends what your concern is, I guess.

Certainly can not repeat the args by lifting it up to a utility function. That's still two instantiations, but those are only on initialisation paths so probably safe.

To actually only instantiate it once means moving it somewhere global. Perhaps that's inside `Config`, and indeed, perhaps the default values for `cache_dir` and `state_dir` should be worked out there, though it looks like `serde` really controls how that object is filled out so I'll need to go read up on.

Otherwise its either passing it down from `main` to where we need it (eww) or some global (`lazy_static`/`once_cell`/etc).

I'll have a run at the config defaults thing, see where we land.

Reply:
New commit added that moves determining default `state_dir` and `cache_dir` into config. I think I like this better, its much simpler code.